lib/cocoapods-core/dependency.rb

```diff
@@ -109,12 +104,8 @@ def initialize(name = nil, *requirements)
         end
 
       elsif requirements.last == :head
-        @head = true
-        requirements.pop
-        unless requirements.empty?
-          raise Informative, 'A `:head` dependency may not specify version ' \
-            "requirements (#{name})."
-        end
+        raise Informative, '`:head` dependencies have been removed. Please use ' \
+          "normal external source dependencies instead of `:head` for `#{name}`."
```

Comment:
That's a great idea @manuyavuz - is it easy to access the `git` repo of the dependency at this point? We could provide a copy & pastable version, if not, I think ^ is :+1: